### PR TITLE
feature(btc_server): signer finalize gRPC endpoint

### PR DESCRIPTION
**Pull request by arminsabouri**

### CODE REVIEW

File: bin/btc-server/src/server.rs [hunk 1]
```diff
@@ -501,14 +489,59 @@ impl rpc::BtcServer for App {
...
+            .collect();
+
+        // Witnesses can be get big. Remove this clone()
+        let witnesses = req.witness;
+        let psbt = self.finalize_signer(outputs_result?, fee_rate, witnesses).map_err(|e| {
+            error!("Failed to finalize signer: {}", e);
+            internal!("Failed to finalize signer: {}", e)
+        })?;
+        let psbt_bytes = hex::decode(psbt.serialize_hex()).map_err(|e| {
+            error!("Failed to serialize psbt: {}", e);
```
Comment (stevenroose) [inline bin/btc-server/src/server.rs]: All internals now print error, so this duplication now is no longer needed.

File: bin/btc-server/src/server.rs [hunk 2]
```diff
@@ -501,14 +489,59 @@ impl rpc::BtcServer for App {
...
+                    .script_pubkey();
+
+                Ok(TxOut { script_pubkey: script_pubkey_result, value: o.value })
+            })
+            .collect();
+
+        // Witnesses can be get big. Remove this clone()
+        let witnesses = req.witness;
+        let psbt = self.finalize_signer(outputs_result?, fee_rate, witnesses).map_err(|e| {
+            error!("Failed to finalize signer: {}", e);
```
Comment (stevenroose) [inline bin/btc-server/src/server.rs]: All internals now print error, so this duplication now is no longer needed.

File: bin/btc-server/src/signer.rs [hunk 3]
```diff
@@ -218,15 +238,47 @@ impl App {
...
+        outputs: Vec<TxOut>,
+        fee_rate: FeeRate,
+        witness: Vec<Vec<u8>>,
+    ) -> Result<Psbt, SigningFinalizeError> {
+        let key_package =
+            self.db.get_key_package()?.ok_or(SigningFinalizeError::MissingKeyPackage)?;
+        let secp_pk = key_package.verifying_key().to_secp_pk().expect("valid pk");
+        let change_script =
+            reth_btc_wallet::address::generate_taproot_change_scriptpubkey(&SECP, &secp_pk);
+        let mut original_psbt = self.make_tx(outputs, fee_rate, change_script).unwrap();
```
Comment (stevenroose) [inline bin/btc-server/src/signer.rs:258]: Uff, this is going to be fragile..

Comment (arminsabouri) [reply bin/btc-server/src/signer.rs:258]: how can we improve?

File: bin/btc-server/src/server.rs (hunk 1 above)
Comment (arminsabouri) [reply bin/btc-server/src/server.rs]: Fixed

File: bin/btc-server/src/server.rs (hunk 2 above)
Comment (arminsabouri) [reply bin/btc-server/src/server.rs]: Fixed

**Review by stevenroose — APPROVED** [review]

**Pull request merged**